### DISCUSSION TITLE
Windows Script to Sync Cloudinit Network Configuration

### DISCUSSION THREAD

**Discussion by ericwang401** [Performave/convoy#62] [Resources]
**This is a repost and wasn't made by me.**

Might help someone here, this pulls the address, gateway, netmask from Proxmox's Cloudinit function in Windows command prompt 
Cloudbase was giving me issues so I prefer to just make a bash script that does it for me automatically

> This script may require a Cloudinit drive to be mounted on the VM where this script is being run on.

```bat
@echo off

setlocal

for /f "tokens=2" %%a in ('type "D:\OPENSTACK\CONTENT\0000" ^| findstr "\<address\>" ^| findstr /r "[0-9.]*"') do set ADDRESS=%%a
for /f "tokens=2" %%a in ('type "D:\OPENSTACK\CONTENT\0000" ^| findstr "\<gateway\>" ^| findstr /r "[0-9.]*"') do set GATEWAY=%%a
for /f "tokens=2" %%a in ('type "D:\OPENSTACK\CONTENT\0000" ^| findstr "\<netmask\>" ^| findstr /r "[0-9\.]*"') do set NETMASK=%%a

echo ADDRESS=%ADDRESS%
echo GATEWAY=%GATEWAY%
echo NETMASK=%NETMASK%

endlocal
```

**Comment by TeYroXOfficial** [comment]
I myself use this version of script which also have setting up DNS'es 

```
@echo off

setlocal

for /f "tokens=2" %%a in ('type "D:\OPENSTACK\CONTENT\0000" ^| findstr "\<address\>" ^| findstr /r "[0-9.]*"') do set ADDRESS=%%a
for /f "tokens=2" %%a in ('type "D:\OPENSTACK\CONTENT\0000" ^| findstr "\<gateway\>" ^| findstr /r "[0-9.]*"') do set GATEWAY=%%a
for /f "tokens=2" %%a in ('type "D:\OPENSTACK\CONTENT\0000" ^| findstr "\<netmask\>" ^| findstr /r "[0-9\.]*"') do set NETMASK=%%a

netsh int ip set address name="Ethernet" source=static ^ address=%ADDRESS% mask=%NETMASK% gate=%GATEWAY%
netsh int ipv4 set dns name="Ethernet" static 8.8.8.8 primary validate=no
netsh int ipv4 add dns name="Ethernet" static 8.8.4.4 index=2

endlocal
```

**Comment by chainlinkhosting** [comment]
Hello,

How is it you get this to work in proxmox?

How would I have it auto run this script from the template startup?

Thanks.

**Reply by ericwang401** [reply]
I haven't used Cloudinit but I would guess you install it and it should work out if the box, no?

**Reply by chainlinkhosting** [reply]
No if you add it in proxmox, It does not work out of the box for windows, Ubuntu has 0 issues with it.
This command prompt is meant to help it work with it, Yet I am unsure how this can be deployed without the need to run the template OS, In away it pre runs so users can login without any issues,

**Reply by chainlinkhosting** [reply]
I want something I don't have to run it manually, So it is all auto on the VM start.

**Reply by ericwang401** [reply]
Got it. @AnushK-Fro could comment on this since he authored this script.
